app/producers/application_producer.rb

```diff
@@ -30,42 +38,29 @@ def logger
     def kafka_ca_cert
       return unless %w[ssl sasl_ssl].include?(Settings.kafka.security_protocol.downcase)
 
-      File.read(Settings.kafka.ssl_ca_location) if Settings.kafka.ssl_ca_location
+      Settings.kafka.ssl_ca_location
     end
 
     def sasl_config
-      return unless Settings.kafka.security_protocol.downcase == 'sasl_ssl'
+      return {} unless Settings.kafka.security_protocol.downcase == 'sasl_ssl'
 
-      config = {
-        sasl_prefix(:username) => Settings.kafka.sasl_username,
-        sasl_prefix(:password) => Settings.kafka.sasl_password,
-        :ssl_ca_certs_from_system => true
+      {
+        'sasl.username' => Settings.kafka.sasl_username,
+        'sasl.password' => Settings.kafka.sasl_password,
+        'sasl.mechanism' => Settings.kafka.sasl_mechanism
```

Comment:
I think we would need to set `security.protocol` as well.
Possible values are `plaintext, ssl, sasl_plaintext, sasl_ssl`, as per (CONFIGURATION.md)[https://github.com/confluentinc/librdkafka/blob/master/CONFIGURATION.md].